### ISSUE TITLE
Do not raise invalid-envvar-default (PLW1508) when wrapped with `int` like in example

### ISSUE THREAD

**Issue by GideonBear** [astral-sh/ruff#27039]
### Summary

[invalid-envvar-default (PLW1508)](https://docs.astral.sh/ruff/rules/invalid-envvar-default/#invalid-envvar-default-plw1508)

```py
import os

int(os.getenv("FOO", 1))
```
It is unclear to me why this is an anti-pattern. There is no confusing behavior here.
I propose to:
* Add an exception for an `os.getenv` call immediately wrapped by `int` where the default argument is an `int` literal
* Change the lint example to for example `os.getenv("FOO", 1)` (without `int` call)

**Comment by ntBre** [comment]
The rule is currently very simple, and I'm not sure it's worth complicating the implementation to check the surrounding context of the `getenv` call. However, I agree with you that the example doesn't seem particularly problematic, so I'd be supportive of a more realistic example where something confusing could actually happen.

**Comment by nightcityblade** [comment]
Hi, I'd like to work on this. I'll update the rule documentation example and submit a PR shortly.
